### ISSUE TITLE
Bug#811572: FTBFS with GCC 6: statement indented as if it were guarded by

### ISSUE THREAD

**Issue by bobek** [troglobit/libite#2]
Hi,

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=811572 was reported with the following content:

This package fails to build with GCC 6.  GCC 6 has not been released
yet, but it's expected that GCC 6 will become the default compiler for
stretch.

Note that only the first error is reported; there might be more.  You
can find a snapshot of GCC 6 in experimental.  To build with GCC 6,
you can set CC=gcc-6 CXX=g++-6 explicitly.

```
> sbuild (Debian sbuild) 0.67.0 (26 Dec 2015) on dl580gen9-02.hlinux
...
> make[1]: Entering directory '/<<PKGBUILDDIR>>'
>   CC      igmp.o
> In file included from defs.h:90:0,
>                  from igmp.c:43:
> libite/lite.h: In function 'fisslashdir':
> libite/lite.h:102:26: error: statement is indented as if it were guarded by... [-Werror=misleading-indentation]
>                           return 0;
>                           ^~~~~~
>
> libite/lite.h:101:4: note: ...this 'if' clause, but it is not
>     if (strlen (dir) > 0) return dir[strlen (dir) - 1] == '/';
>     ^~
>
> cc1: all warnings being treated as errors
> Makefile:51: recipe for target 'igmp.o' failed
> make[1]: *** [igmp.o] Error 1
```

David Malcolm was so nice, that he followed up with 

>(I'm the upstream gcc author of -Wmisleading-indentation)
>
>Looking at,
 > https://github.com/troglobit/libite/blob/master/lite.h#L134
>the code in question seems to be:

```
static inline int fisslashdir(char *dir)
{
   if (!dir)             return 0;
   if (strlen (dir) > 0) return dir[strlen (dir) - 1] == '/';
                         return 0;
}
```

**Comment by troglobit** [comment]
Thanks @bobek for the report!

At this point in time I cannot make heads or tails of that error, since syntactically I see nothing wrong with the code?

I'm a bit surprised Debian is making `-Werror=misleading-indentation` default with GCC-6. It may be a help to maintainers, but for building packages for general consumption? 

I'm sorry Antonin, I don't want to lay it all on you, I'm sure Debian has a good reason for this policy. I'll try to get my hands on a GCC-6 compiler, even though I'm a bit hesitant to this Python-style magic applied to C ;-)

**Comment by troglobit** [comment]
OK, now I get it ... I've added `-Werror` to [pimd](https://github.com/troglobit/pimd) and GCC6 adds `-Wmisleading-indentation`!

Bad form by me, `-Werror` should only be used by me as a maintainer-type flag.

**Comment by troglobit** [comment]
There, fixed! :)

**Comment by troglobit** [comment]
@bobek For an easy fix to Debian, before the upcoming pimd v2.3.2 release, cherry-pick troglobit/pimd@9ca8e4c as a quilt patch.

**Comment by bobek** [comment]
Yeap. I have tested compilation under gcc-6 and it works. It will just complain about indentation on multiple other occurrences. I've pushed the patched version for now (as Debian version 2.3.1-2).

**Comment by troglobit** [comment]
Good, thanks again @bobek :+1:
